bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/Activator.java

```diff
@@ -32,4 +54,33 @@ public void stop(BundleContext context) throws Exception {
     public static BundleContext getContext() {
         return Activator.context;
     }
+
+    /**
+     * Returns the {@link ThingTypeRegistry}
+     *
+     * @return {@link ThingTypeRegistry} or null
+     */
+    public static ThingTypeRegistry getThingTypeRegistry() {
+        return Activator.thingTypeRegistry;
+    }
+
+    private ServiceBinder channelTypeServiceBinder;
+
+    private ServiceBinder thingTypeServiceBinder;
+
+    @Override
+    public void start(BundleContext context) throws Exception {
+        Activator.context = context;
+        thingTypeServiceBinder = new ServiceBinder(context, new ThingTypeRegistryBinder());
+        thingTypeServiceBinder.open();
+        channelTypeServiceBinder = new ServiceBinder(context, new ChannelTypeRegistryBinder());
+        channelTypeServiceBinder.open();
+    }
+
+    @Override
+    public void stop(BundleContext context) throws Exception {
+        channelTypeServiceBinder.close();
```

Comment:
It depends. In that case it is just an internal variable. So the only object which can have a reference is the object itself. The normal JVM garbage collector will take care about this, when the service itself will be released.

But nevertheless I added setting it to null, as a lot of people are always confused about that and expect it to be an error.